### DISCUSSION TITLE
Sabin Adams - Prisma

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#164] [Announcements]
- https://twitter.com/sabinthedev
- https://www.sabinthedev.com
- https://github.com/sabinadams
- https://www.linkedin.com/in/sabin-adams-656261122/

## Topics

- Prisma
- Remix
- E2E Type Safety
- Edge databases

## Questions

- [x] Where did you get started on your developer journey and how did you end up at prisma?

## Prisma

- [x] what is prisma and why would I want to use something like that? What's the alternative?
- [x] How do you define DBs in prisma? How do migrations happen?
- [x] Prisma can be backed by a bunch of different DBs. What the differences between using them? What your personal favorite?
- [x] What things does prisma do to improve the DX of my database? (IDE or otherwise)
- [x] What is prisma accelerate?

## E2E Type Safety

- [x] What is your preferred stack for achieving e2e type safety?
- [x] Have you used things like tRPC?
- [x] What challenges come about when trying to achieve this? 
- [x] How do you feel about the tradeoffs of code generation vs more native typescript friendly APIs? 

## Future

- [x] Where do you see the future of full stack applications going? Do you think more tools or new tools are needed to bridge the gap?

**Comment by sabinadams** [comment]
https://marketplace.visualstudio.com/items?itemName=wicked-labs.sequoia

https://www.swyx.io/self-provisioning-runtime

https://www.amazon.com/Canson-Drawing-Acrylic-Gouache-Pastels/dp/B01H33DHEM/ref=sr_1_2?hvadid=598609479783&hvdev=c&hvlocphy=9031708&hvnetw=g&hvqmt=e&hvrand=4762151141273890168&hvtargid=kwd-438307667193&hydadcr=4834_13229512&keywords=canson%2Bblack%2Bsketchbook&qid=1683235901&sr=8-2&th=1

https://www.amazon.com/Sakura-Gelly-Moonlight-Assorted-Colors/dp/B000GZOCA8/ref=sr_1_2?hvadid=616931511259&hvdev=c&hvlocphy=9031708&hvnetw=g&hvqmt=e&hvrand=4694435914658466485&hvtargid=kwd-151587091265&hydadcr=24662_13611802&keywords=jelly%2Broll%2Bpen&qid=1683235916&sr=8-2&th=1

**Comment by hipstersmoothie** [comment]
https://www.npmjs.com/package/openai/v/3.2.1
https://vercel.com/docs/storage/vercel-postgres

**Comment by just-be-dev** [comment]
- https://scrapscript.org/
- https://www.cs.cornell.edu/~asampson/blog/flattening.html
